definitions/features/foreman_proxy.rb

```diff
@@ -0,0 +1,90 @@
+class Features::ForemanProxy < ForemanMaintain::Feature
+  metadata do
+    label :foreman_proxy
+  end
+
+  class DhcpCurlResponse
+    attr_reader :result, :http_code, :error_msg
+
+    def initialize(response, http_code, http_error_msg)
+      @result = response || ''
+      @http_code = http_code
+      @error_msg = generate_error_msg(http_error_msg)
+    end
+
+    private
+
+    def generate_error_msg(msg_string)
+      <<-EOF
+      \n#{msg_string} #{http_code}. Response: #{result.inspect}
+      EOF
+    end
+  end
+
+  attr_reader :dhcpd_conf_file, :cert_path
+
+  def initialize
+    @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
+    @cert_path = ForemanMaintain.config.foreman_proxy_cert_path
+  end
+
+  def valid_dhcp_configs?
+    dhcp_req_pass? && !syntax_error_exists?
+  end
+
+  private
+
+  def dhcp_curl_cmd
+    "curl -w '\n%{http_code}' -slient -ks --cert #{cert_path}/client_cert.pem \
+      --key #{cert_path}/client_key.pem \
+      --cacert #{cert_path}/proxy_ca.pem https://$(hostname):9090/dhcp"
+  end
+
+  def find_http_error_msg(array_output, curl_http_status)
+    http_line = ''
+    array_output.each do |str|
+      next unless str.include?('HTTP')
+      http_line = str
+    end
+    http_line.split(curl_http_status.to_s).last.strip
+  end
+
+  def run_dhcp_curl
+    curl_resp = execute(dhcp_curl_cmd)
+    array_output = curl_resp.split(/\r\n/)
+    result_array = array_output.last.split(/\n/)
+    curl_http_status = result_array.delete_at(result_array.length - 1).strip.to_i
+    curl_http_resp = json_parse(result_array.join(''))
+    DhcpCurlResponse.new(
```

Comment:
Logic is okay. Just ought to keep features clean and to its purpose.
`DhcpCurlResponse` can be `Utils::Dhcp`